src/settings/components/MCLPaginationFields.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import { Field } from 'react-final-form-html5-validation';
+
+import {
+  Col,
+  Layout,
+  Row,
+  TextField,
+} from '@folio/stripes/components';
+
+const lists = ['agreementLines', 'agreementEresources', 'entitlementOptions', 'packageContents', 'entitlements'];
+
+const inRange = (x, min, max) => {
+  return ((x - min) * (x - max) <= 0);
+};
+
+const validate = (fieldValue, min, max) => {
+  return (!inRange(fieldValue, min, max)) ?
+    <FormattedMessage id="ui-agreements.settings.error.valueNotInRange" values={{ min, max }} /> : undefined;
+};
+
+const MCLPaginationFields = () => {
+  return (
+    <>
+      <strong>
+        <Layout className="padding-bottom-gutter padding-top-gutter" data-test-mcl-description>
+          <FormattedMessage id="ui-agreements.settings.mcl.pagination.description" />
+        </Layout>
+        <Layout className="padding-bottom-gutter">
+          <Row>
+            <Col xs={8}> <FormattedMessage id="ui-agreements.settings.mcl" /> </Col>
+            <Col xs={4}> <FormattedMessage id="ui-agreements.settings.mcl.pageSize" /> </Col>
+          </Row>
+        </Layout>
+      </strong>
+      { lists.map((mcl, index) => (
+        <Row key={`row-${index}`}>
+          <Col xs={8}>
+            <FormattedMessage id={`ui-agreements.settings.mcl.${mcl}`} />
+          </Col>
+          <Col xs={4}>
+            <Field
```

Comment:
You dont need the `Field` from `react-final-form-html5-validation` for this, as you have your own custom validate logic here. You could use out of the box validators such as `rangeOverflow` and `rangeUnderflow` like how we use in the `CustomPropertyValue` component, and in that case importing `Field` from `react-final-form-html5-validation` makes sense